test/animators/baseAnimatorTests.ts

```diff
@@ -0,0 +1,156 @@
+        assert.strictEqual(actualTotalTime, expectedTotalTime,
+          "Animation should not exceed the maxTotalDuration time constraint when step duration set too high");
+      });
+
+      it("maxTotalDuration() is just a constraint", () => {
+        var iterationSteps = 2;
+        var startDelay = 0;
+        var stepDuration = 100;
+        var iterativeDelay = 100;
+        var maxTotalDuration = 5000;
+
+        var expectedTotalTime = 200;
+
+        var animator = new Plottable.Animators.Base();
+        animator.startDelay(startDelay);
+        animator.stepDuration(stepDuration);
+        animator.iterativeDelay(iterativeDelay);
+        animator.maxTotalDuration(maxTotalDuration);
+
+        var actualTotalTime = animator.totalTime(iterationSteps);
+        assert.strictEqual(actualTotalTime, expectedTotalTime,
+          "The total duration constraint is just an upper bound");
+      });
+
+      it("maxTotalDuration() edge case for 1 step", () => {
+        var iterationSteps = 1;
+        var startDelay = 0;
+        var stepDuration = 333;
+        var iterativeDelay = 432;
+
+        var animator = new Plottable.Animators.Base();
+        animator.startDelay(startDelay);
+        animator.stepDuration(stepDuration);
+        animator.iterativeDelay(iterativeDelay);
+
+        var actualTotalTime = animator.totalTime(iterationSteps);
+        assert.strictEqual(actualTotalTime, stepDuration,
+          "The total time is just the time for one step");
+      });
+
+      it("maxTotalDuration() edge case for 1 step and startDelay", () => {
+        var iterationSteps = 1;
+        var startDelay = 213;
+        var stepDuration = 333;
+        var iterativeDelay = 432;
+
+        var expectedTotalTime = startDelay + stepDuration;
+
+        var animator = new Plottable.Animators.Base();
+        animator.startDelay(startDelay);
+        animator.stepDuration(stepDuration);
+        animator.iterativeDelay(iterativeDelay);
+
+        var actualTotalTime = animator.totalTime(iterationSteps);
+        assert.strictEqual(actualTotalTime, expectedTotalTime,
+          "Total time is the time for one step, plus waiting for the start delay");
+      });
+
+      it("_getAdjustedIterativeDelay() works with maxTotalDuration constraint", () => {
+        var iterationSteps = 2;
+        var startDelay = 0;
+        var stepDuration = 1000;
+        var iterativeDelay = 1000000;
+        var maxTotalDuration = 1500;
+
+        var animator = new Plottable.Animators.Base();
+        animator.startDelay(startDelay);
+        animator.stepDuration(stepDuration);
+        animator.iterativeDelay(iterativeDelay);
+        animator.maxTotalDuration(maxTotalDuration);
+
+        var expectedIterativeDelay = 500;
+        // 1 |  ###########
+        // 2 |       ###########
+        //   +------------------------
+        //   |  |    |    |    |    |
+        //     0.0  0.5  1.0  1.5  2.0
```

Comment:
Is there a way we can get this idea through without pictures.  Pictures take up space :/